### PR TITLE
Implement HU-Locking

### PR DESCRIPTION
**Pull request by av-ps**

### CODE REVIEW

**Review by metas-ts — APPROVED** [review]

File: misc/de-metas-common/de-metas-common-manufacturing/src/main/java/de/metas/common/handlingunits/JsonClearanceStatusInfo.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+
+@Value
+@Builder
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY, getterVisibility = JsonAutoDetect.Visibility.NONE, isGetterVisibility = JsonAutoDetect.Visibility.NONE, setterVisibility = JsonAutoDetect.Visibility.NONE)
+@JsonDeserialize(builder = JsonClearanceStatusInfo.JsonClearanceStatusInfoBuilder.class)
```
Comment (teosarca) [inline misc/de-metas-common/de-metas-common-manufacturing/src/main/java/de/metas/common/handlingunits/JsonClearanceStatusInfo.java:36]: follow-up: use `@Jacksonized`
e.g. https://github.com/metasfresh/metasfresh/blob/2377ad7d619e00aaa249e57050a1c48e294e808f/backend/de.metas.workflow.rest-api/src/main/java/de/metas/workflow/rest_api/activity_features/set_scanned_barcode/JsonQRCode.java#L10

cc @metas-ts

File: misc/de-metas-common/de-metas-common-manufacturing/src/main/java/de/metas/common/handlingunits/JsonSetClearanceStatusRequest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+
+import javax.annotation.Nullable;
+
+@Value
+@Builder
```
Comment (teosarca) [inline misc/de-metas-common/de-metas-common-manufacturing/src/main/java/de/metas/common/handlingunits/JsonSetClearanceStatusRequest.java:33]: follow-up: use `@Jacksonized`
e.g. https://github.com/metasfresh/metasfresh/blob/2377ad7d619e00aaa249e57050a1c48e294e808f/backend/de.metas.workflow.rest-api/src/main/java/de/metas/workflow/rest_api/activity_features/set_scanned_barcode/JsonQRCode.java#L10

cc @metas-ts

File: misc/services/camel/de-metas-camel-externalsystems/de-metas-camel-grssignum/src/main/java/de/metas/camel/externalsystems/grssignum/to_grs/api/model/JsonHUClear.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+
+import javax.annotation.Nullable;
+
+@Value
+@JsonDeserialize(builder = JsonHUClear.JsonHUClearBuilder.class)
```
Comment (teosarca) [inline misc/services/camel/de-metas-camel-externalsystems/de-metas-camel-grssignum/src/main/java/de/metas/camel/externalsystems/grssignum/to_grs/api/model/JsonHUClear.java:36]: follow-up: use `@Jacksonized`
e.g. https://github.com/metasfresh/metasfresh/blob/2377ad7d619e00aaa249e57050a1c48e294e808f/backend/de.metas.workflow.rest-api/src/main/java/de/metas/workflow/rest_api/activity_features/set_scanned_barcode/JsonQRCode.java#L10

cc @metas-ts

File: misc/de-metas-common/de-metas-common-manufacturing/src/main/java/de/metas/common/handlingunits/JsonAllowedHUClearanceStatuses.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+
+import java.util.List;
+
+@Value
+@Builder
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY, getterVisibility = JsonAutoDetect.Visibility.NONE, isGetterVisibility = JsonAutoDetect.Visibility.NONE, setterVisibility = JsonAutoDetect.Visibility.NONE)
+@JsonDeserialize(builder = JsonAllowedHUClearanceStatuses.JsonAllowedHUStatusesBuilder.class)
```
Comment (teosarca) [inline misc/de-metas-common/de-metas-common-manufacturing/src/main/java/de/metas/common/handlingunits/JsonAllowedHUClearanceStatuses.java:38]: follow-up: use Jacksonized...
cc @metas-ts

**Pull request merged**